pkg/controller/privatelink/privatelink_controller.go

```diff
@@ -0,0 +1,486 @@
+		return err
+	}
+
+	// Watch for changes to ClusterProvision
+	if err := c.Watch(source.Kind(mgr.GetCache(), &hivev1.ClusterProvision{}),
+		handler.EnqueueRequestForOwner(mgr.GetScheme(), mgr.GetRESTMapper(), &hivev1.ClusterDeployment{}, handler.OnlyControllerOwner())); err != nil {
+		logger.WithError(err).Error("error watching cluster provision")
+		return err
+	}
+
+	// Watch for changes to ClusterDeprovision
+	if err := c.Watch(source.Kind(mgr.GetCache(), &hivev1.ClusterDeprovision{}),
+		handler.EnqueueRequestForOwner(mgr.GetScheme(), mgr.GetRESTMapper(), &hivev1.ClusterDeployment{}, handler.OnlyControllerOwner())); err != nil {
+		logger.WithError(err).Error("error watching cluster deprovision")
+		return err
+	}
+
+	return nil
+}
+
+func (r *ReconcilePrivateLink) Reconcile(ctx context.Context, request reconcile.Request) (result reconcile.Result, returnErr error) {
+	logger := controllerutils.BuildControllerLogger(ControllerName, "clusterDeployment", request.NamespacedName)
+	logger.Debug("reconciling cluster deployment")
+	recobsrv := hivemetrics.NewReconcileObserver(ControllerName, logger)
+	defer recobsrv.ObserveControllerReconcileTime()
+
+	if r.hubActuator == nil {
+		logger.Error("hub actuator is not set")
+		return reconcile.Result{}, errors.New("hub actuator is not set")
+	}
+
+	// Fetch the ClusterDeployment instance
+	cd := &hivev1.ClusterDeployment{}
+	err := r.Get(context.TODO(), request.NamespacedName, cd)
+	if apierrors.IsNotFound(err) {
+		logger.Debug("cluster deployment not found")
+		return reconcile.Result{}, nil
+	}
+	if err != nil {
+		// Error reading the object - requeue the request.
+		logger.WithError(err).Error("error getting ClusterDeployment")
+		return reconcile.Result{}, err
+	}
+	logger = controllerutils.AddLogFields(controllerutils.MetaObjectLogTagger{Object: cd}, logger)
+
+	if paused, err := strconv.ParseBool(cd.Annotations[constants.ReconcilePauseAnnotation]); err == nil && paused {
+		logger.Info("skipping reconcile due to ClusterDeployment pause annotation")
+		return reconcile.Result{}, nil
+	}
+
+	// Initialize cluster deployment conditions if not present
+	newConditions, changed := conditions.InitializeConditions(cd)
+	if changed {
+		cd.Status.Conditions = newConditions
+		logger.Info("initializing private link controller conditions")
+		if err := r.Status().Update(context.TODO(), cd); err != nil {
+			logger.WithError(err).Log(controllerutils.LogLevel(err), "failed to update cluster deployment status")
+			return reconcile.Result{}, err
+		}
+		return reconcile.Result{}, nil
+	}
+
+	var clientActuator Actuator
+	var privateLinkEnabled bool
+	switch {
+	case cd.Spec.Platform.AWS != nil:
+		if cd.Spec.Platform.AWS.PrivateLink == nil {
+			logger.Debug("controller cannot service the clusterdeployment, so skipping")
+			return reconcile.Result{}, nil
+		}
+		privateLinkEnabled = cd.Spec.Platform.AWS.PrivateLink.Enabled
+		clientActuator, err = r.GetActuator(configv1.AWSPlatformType, logger)
```

Comment:
The (now) spokeActuator is specific to each individual CD, having cloud-specific code and client service account credentials. We could create actuators for each cloud and pass around a client instead.

Reply:
As discussed, two actuators:
- Hub actuator for the RH-owned infra that hive is running *on*. Only AWS supported for the current pass.
- Spoke actuator tied to the cloud the spoke is running on, with *two cloud clients*:
  - RH-owned infra for the "link"
  - Customer-owned infra for the, uhh, "endpoint"?

...and we can't do what I suggested here because the reconciler is a singleton for the controller instance, and sharing it among CDs would be Bad™.